crates/node-core/src/node_config.rs

```diff
@@ -768,7 +770,25 @@ impl NodeConfig {
                 self.network.discovery.port + self.instance - 1,
             ));
 
-        cfg_builder.build(client)
+        let config = cfg_builder.build(client);
+
+        if !self.network.discovery.enable_discv5_discovery {
+            return config
+        }
+        // work around since discv5 config builder can't be integrated into network config builder
+        // due to unsatisfied trait bounds
+        config.discovery_v5_with_config_builder(|builder| {
```

Comment:
why is that?

Reply:
serialise and deserialise trait isn't implemented for discv5